numpy/core/arrayprint.py

```diff
@@ -285,7 +268,27 @@ def _get_format_function(data, precision, suppress_small, formatter):
             if key in fkeys:
                 formatdict[key] = formatter[key]
 
+    return formatdict
+
+def _get_format_function(data, precision, suppress_small, formatter):
+    """
+    find the right formatting function for the dtype_
+    """
+    dtype_ = data.dtype
+    if dtype_.fields is not None:
+        format_functions = []
+        for field_name in dtype_.names:
+            field_values = data[field_name]
+            is_array_field = 1 < field_values.ndim
+            format_function = _get_format_function(
+                    ravel(field_values), precision, suppress_small, formatter)
+            if is_array_field:
+                format_function = SubArrayFormat(format_function)
```

Comment:
Just a note: `SubarrayFormat` here does not take into account `precision`, `suppress_small`, `formatter`. Maybe it should, but if so let's leave that for a future PR. Current numpy master doesn't take them into account either.